setup.py

```diff
@@ -66,53 +85,83 @@ def run_tests(self):
             args = shlex.split(self.tox_args)
+
+        LibDDWaf_Download.download_dynamic_library()
         errno = tox.cmdline(args=args)
         sys.exit(errno)
 
 
-class CMake(BuildExtCommand):
-    def build_extension(self, ext):
-        import shutil
-        import subprocess
-        import tempfile
-
-        to_build = set()
-        # Detect if any source file sits next to a CMakeLists.txt file
-        for source in ext.sources:
-            source_dir = os.path.dirname(os.path.realpath(source))
-            if os.path.exists(os.path.join(source_dir, "CMakeLists.txt")):
-                to_build.add(source_dir)
+class LibDDWaf_Download(BuildPyCommand):
+    @staticmethod
+    def download_dynamic_library():
+        TRANSLATE_SUFFIX = {"Windows": ".dll", "Darwin": ".dylib", "Linux": ".so"}
+        AVAILABLE_RELEASES = {
+            "Windows": ["win32", "x64"],
+            "Darwin": ["arm64", "x86_64"],
+            "Linux": ["aarch64", "x86_64"],
+        }
+        SUFFIX = TRANSLATE_SUFFIX[CURRENT_OS]
 
-        if not to_build:
-            # Build the extension as usual
-            BuildExtCommand.build_extension(self, ext)
+        if os.path.isdir(LIBDDWAF_DOWNLOAD_DIR) and len(os.listdir(LIBDDWAF_DOWNLOAD_DIR)):
             return
 
-        try:
-            cmake_command = os.environ.get("CMAKE_COMMAND", "cmake")
-            build_type = "RelWithDebInfo" if DEBUG_COMPILE else "Release"
-            opts = ["-DCMAKE_BUILD_TYPE={}".format(build_type)]
-            if platform.system() == "Windows":
-                opts.extend(["-A", "x64" if platform.architecture()[0] == "64bit" else "Win32"])
-            else:
-                opts.extend(["-G", "Ninja"])
-                ninja_command = os.environ.get("NINJA_COMMAND", "")
-                if ninja_command:
-                    opts.append("-DCMAKE_MAKE_PROGRAM={}".format(ninja_command))
-
-            for source_dir in to_build:
-                try:
-                    build_dir = tempfile.mkdtemp()
-                    subprocess.check_call([cmake_command, "-S", source_dir, "-B", build_dir] + opts)
-                    subprocess.check_call([cmake_command, "--build", build_dir, "--config", build_type])
-                finally:
-                    if not DEBUG_COMPILE:
-                        shutil.rmtree(build_dir, ignore_errors=True)
-
-            BuildExtCommand.build_extension(self, ext)
-        except Exception as e:
-            print('WARNING: building extension "%s" failed: %s' % (ext.name, e))
-            raise
+        if not os.path.isdir(LIBDDWAF_DOWNLOAD_DIR):
+            os.makedirs(LIBDDWAF_DOWNLOAD_DIR)
+
+        build_platform = get_build_platform()
+        for arch in AVAILABLE_RELEASES[CURRENT_OS]:
+            if CURRENT_OS == "Darwin" and build_platform.endswith("x86_64") and arch == "arm64":
+                continue
+
+            arch_dir = os.path.join(LIBDDWAF_DOWNLOAD_DIR, arch)
+
+            if os.path.isdir(arch_dir):
```

Comment:
if we have the dir does it mean we already have the correct/valid/up-to-date files?

Reply:
We assume the files are correct, valid and up to date if the directory exists, to avoid downloading them every time, as they are needed for more than one command